tests/test_storageClass.py

```diff
@@ -0,0 +1,86 @@
+#
+# LSST Data Management System
+#
+# Copyright 2018  AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+import unittest
+import lsst.utils.tests
+
+import lsst.daf.butler.core.storageClass as storageClass
+
+"""Tests related to the StorageClass infrastructure.
+"""
+
+
+class PythonType:
+    """A dummy class to test the registory of Python types."""
+    pass
+
+
+class StorageClassFactoryTestCase(lsst.utils.tests.TestCase):
+    """Tests of the storage class infrastructure.
+    """
+
+    def testCreation(self):
+        """Test that we can dynamically create storage class subclasses.
+
+        This is critical for testing the factory functions."""
+        className = "TestImage"
+        newclass = storageClass.makeNewStorageClass(className, dict, None)
+        sc = newclass()
+        self.assertIsInstance(sc, storageClass.StorageClass)
+        self.assertEqual(sc.name, className)
+        self.assertIsNone(sc.components)
+        self.assertEqual(sc.type, dict)
+
+        # Check that this class is listed in the subclasses
+        self.assertIn(className, newclass.subclasses)
```

Comment:
I might leave it in for a little just in case I have a problem where I need a storage class that has been made concrete somewhere else but is not in this registry.